hyperspy/misc/array_tools.py

```diff
@@ -470,8 +470,9 @@ def numba_closest_index_round(axis_array,value_array):
     index_array = np.empty_like(value_array,dtype='uint')
     #assign on flat, iterate on flat.
     for i,v in enumerate(value_array.flat):
-        vdiff_array = np.abs(axis_array - v)
-        index_array.flat[i] = np.flatnonzero(vdiff_array == np.min(vdiff_array))[-1]
+        rtol=1e-10
+        machineepsilon = np.min(np.diff(axis_array))*rtol
+        index_array.flat[i] = np.abs(axis_array - v + machineepsilon).argmin()
```

Comment:
Idk if that's why the tests fail, but probably you need to replace:  `machineepsilon = np.min(np.abs(np.diff(axis_array)))*rtol` 

Also, I'd suggest you put this outside for loop for efficiency. this does not depend on value and need to be computed only once

Reply:
`macineepsilon` has to be of course added to `v` and not to `axis_array`. so `- machineepsilon`. I also differentiate now between positive and negative values to round away from zero (otherwise we would have rounded up `-0.5`).